### DISCUSSION TITLE
Porting my Nb 8.0.2 RCP, apache ANT for Maven

### DISCUSSION THREAD

**Discussion by guilherme-oliveria** [apache/netbeans#4468] [General]
### Apache NetBeans version

Apache NetBeans latest daily build

### What happened

Hello, I have a legacy application using netbeans RCP platform in IDE 8.0.2, which is built by ANT, and I am migrating to Maven. I followed these steps: https://netbeans.apache.org/tutorials/80/nbm-maven-commandline.html,
using NBM version 4.1, <groupId>org.codehaus.mojo</groupId> <groupId>org.apache.netbeans.utilities</groupId> gives an internal error in NB 8.0.2. I used the same dependencies that the RCP platform uses in the 802 version, they can be found here
http://netbeans.apidesign.org/maven2/org/netbeans/api/org-openide-awt/ . org-openide-* <version> 802.

the Build with mvn clean install is a success, the application also runs on the default project of using mvn nbm:run-platform
```
<modules>
    <module>branding</module>
    <module>application</module>
</modules>
```

but if I add my modules, the maven build is successful and using the same dependencies as when it was ANT the error in org-openide-*

```
<modules>
    <module>branding</module>
    <module>application</module>
    <module>myModule</module>
</modules>
```

ERROR:
```
[INFO] java.lang.Error
[INFO] 	at javax.swing.UIDefaults.getUIError(UIDefaults.java:731)
[INFO] 	at javax.swing.MultiUIDefaults.getUIError(MultiUIDefaults.java:130)
[INFO] 	at javax.swing.UIDefaults.getUI(UIDefaults.java:761)
[INFO] 	at javax.swing.UIManager.getUI(UIManager.java:1016)
[INFO] 	at com.jidesoft.swing.StyledLabel.updateUI(Unknown Source)
[INFO] 	at javax.swing.JLabel.<init>(JLabel.java:164)
[INFO] 	at br.com.petrobras.rcp.jide.tabela.TableManager.ajustarDimensionamentoColunas(TableManager.java:503) **
[INFO] 	at br.com.petrobras.jconsuelo.clientercp.processo.ui.ConsultarCenarioPanel.initTable(ConsultarCenarioPanel.java:220)
[INFO] 	at br.com.petrobras.jconsuelo.clientercp.processo.ui.ConsultarCenarioPanel.<init>(ConsultarCenarioPanel.java:105)
[INFO] 	at br.com.petrobras.jconsuelo.clientercp.processo.cookie.ext.ConsultarCenarioCookieImpl.consultarCenario(ConsultarCenarioCookieImpl.java:58)
[INFO] 	at br.com.petrobras.jconsuelo.clientercp.processo.action.cenario.ConsultarCenarioAction.actionPerformed(ConsultarCenarioAction.java:50)
[INFO] 	at org.openide.awt.InjectorExactlyOne.actionPerformed(InjectorExactlyOne.java:78)
[INFO] 	at org.openide.awt.ContextAction$Performer.actionPerformed(ContextAction.java:226)
[INFO] 	at org.openide.awt.ContextManager.actionPerformed(ContextManager.java:260)
[INFO] 	at org.openide.awt.ContextAction.actionPerformed(ContextAction.java:109)
[INFO] 	at org.openide.util.actions.ActionInvoker$1.run(ActionInvoker.java:93)
[INFO] 	at org.openide.util.actions.ActionInvoker.doPerformAction(ActionInvoker.java:116)
[INFO] 	at java.awt.EventDispatchThread.pumpEventsForHierarchy(EventDispatchThread.java:105)
[INFO] 	at java.awt.EventDispatchThread.pumpEvents(EventDispatchThread.java:101)
[INFO] 	at java.awt.EventDispatchThread.pumpEvents(EventDispatchThread.java:93)
[INFO] 	at java.awt.EventDispatchThread.run(EventDispatchThread.java:82)
[INFO] UIDefaults.getUI() failed: no ComponentUI class for: com.jidesoft.grid.AutoFilterTableHeader$0[,0,0,0x0,invalid,alignmentX=0.0,alignmentY=0.0,border=,flags=0,maximumSize=,minimumSize=,preferredSize=,defaultIcon=,disabledIcon=,disabledSelectedIcon=,margin=null,paintBorder=true,paintFocus=true,pressedIcon=,rolloverEnabled=false,rolloverIcon=,rolloverSelectedIcon=,selectedIcon=,text=,defaultCapable=true]
[
OR
[INFO] UIDefaults.getUI() failed: no ComponentUI class for: javax.swing.JMenu[,0,0,0x0,invalid,alignmentX=0.0,alignmentY=0.0,border=,flags=0,maximumSize=,minimumSize=,preferredSize=,defaultIcon=,disabledIcon=,disabledSelectedIcon=,margin=null,paintBorder=false,paintFocus=false,pressedIcon=,rolloverEnabled=false,rolloverIcon=,rolloverSelectedIcon=,selectedIcon=,text=]
[INFO] java.lang.Error
OR
[INFO] UIDefaults.getUI() failed: no ComponentUI class for: javax.swing.JPopupMenu$Separator[,0,0,0x0,invalid,alignmentX=0.0,alignmentY=0.0,border=,flags=0,maximumSize=,minimumSize=,preferredSize=,orientation=HORIZONTAL]
[INFO] java.lang.Error

[INFO] UIDefaults.getUI() failed: no ComponentUI class for: com.jidesoft.swing.ClickThroughStyledLabel[,0,0,0x0,invalid,alignmentX=0.0,alignmentY=0.0,border=,flags=0,maximumSize=,minimumSize=,preferredSize=,defaultIcon=,disabledIcon=,horizontalAlignment=LEADING,horizontalTextPosition=TRAILING,iconTextGap=4,labelFor=,text=,verticalAlignment=CENTER,verticalTextPosition=CENTER]
[INFO] java.lang.Error
```

I'm not migrating in version just changing build from ANT to Maven and using the same libs, but UIDefaults.getUI() fails on the same components.
here is a similar error but for macOS user
https://bz.apache.org/netbeans/show_bug.cgi?id=182507
would need a light, with Build ANT the same java8 version works

``

### Did this work correctly in an earlier version?

Apache NetBeans 12.3 or earlier

### Operating System

Windows 64bits

### JDK

1.8.0_281

### Apache NetBeans packaging

Apache NetBeans platform

### Are you willing to submit a pull request?

Yes

### Code of Conduct

Yes

**Comment by neilcsmith-net** [comment]
I don't see an issue at present in NetBeans from this, although there may be - moving to discussions for now.

**Comment by guilherme-oliveria** [comment]
My problem was because of com.jidesoft dependencies importing directly in each module doing repetition, so I created a pom.xml directory of type <packaging>nbm</packaging> and put the jide dependencies there, and where I imported the jar , I started to import my new nbm project

`        <dependency>
            <groupId>com.jidesoft</groupId>
            <artifactId>jide-grids</artifactId>
        </dependency>
        <dependency>
            <groupId>com.jidesoft</groupId>
            <artifactId>jide-common</artifactId>
        </dependency>
        <dependency>
            <groupId>com.jidesoft</groupId>
            <artifactId>jide-properties</artifactId>
        </dependency>
        <dependency>
            <groupId>com.jidesoft</groupId>
            <artifactId>jide-components</artifactId>
        </dependency>`
